src/Microsoft.AspNet.Mvc.TagHelpers/Internal/JavaScriptStringArrayEncoder.cs

```diff
@@ -1,43 +0,0 @@
-// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System.IO;
-using System.Text.Encodings.Web;
-
-namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
-{
-    /// <summary>
-    /// Methods for encoding <c>string[]</c>s for use as a JavaScript array literal.
-    /// </summary>
-    public static class JavaScriptStringArrayEncoder
-    {
-        /// <summary>
-        /// Encodes a <c>string[]</c> for safe use as a JavaScript array literal in many contexts, including
-        /// inline in an HTML file.
-        /// </summary>
-        public static string Encode(JavaScriptEncoder encoder, string[] values)
```

Comment:
The earlier code modified the list in place (https://github.com/aspnet/Mvc/pull/3923/files#diff-5574912c61dd39d9a17a1c579ae2a7c0L322) based on the value of `AppendVersion`. Since we return the cached `IReadOnlyList`, I changed the code to calculate the fallBackUrl in place and encode it. Passing `AppendVersion` + `IFileVersionProvider` seems weird. I could refactor this bit out in to a separate method in `LinkTagHelper` if it's easier to read.